### PR TITLE
Fix #69 (both parts)

### LINKED ISSUE #69: error building
1. No such file or directory: 'numpy-atlas.dll'
helps get C:\Python27\Lib\site-packages\numpy\core\numpy-atlas.dll and copied it to the same folder that has the setup.py
2.  not building with py2exe-0.9.2.2.win32 on python 2.7
helps old py2exe-0.6.9.win32-py2.7
3. on x64 not working 7z becose C:\Program Files (x86) folder
4. bad problem not building 7z and sfx
svn wiki not found

when many songs crush when scanning songs. 
.fofix-cache delete, edit .inis not help
[    6.344000] (E) Terminating due to unhandled exception: 
  File "fofix\game\song\song.pyo", line 3397, in updateSongDatabase
  File "fofix\game\song\song.pyo", line 3010, in getAvailableSongs
  File "fofix\game\song\song.pyo", line 264, in __init__
  File "fofix\game\song\song.pyo", line 329, in getScores
TypeError: Non-hexadecimal digit found

### PR DESCRIPTION
**Pull request by wrzwicky**
Fix issue #69 (both parts). Also finish py2exe for win32.

Rebase of #87 to current fofix/master, plus better commits.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,6 +19,7 @@ fofix/core/VideoPlayer/_VideoPlayer.c
 fofix/core/pypitch/_pypitch.cpp
 bin
 build
+dist
```
Comment (Linkid) [inline .gitignore]: You already ignore it in b06b116. Remove the extra line, then ;).

File: fofix/game/song/song.py [hunk 2]
```diff
@@ -325,20 +325,31 @@ def getScores(self, part):
             scores = self._get("scores", str, "")
             scores_ext = self._get("scores_ext", str, "")
 
+        if scores:
+            try:
+                scores = cerealizer.loads(binascii.unhexlify(scores))
+            except:
```
Comment (Linkid) [inline fofix/game/song/song.py]: You should specify the exception (use `Exception` instead).

File: fofix/game/song/song.py [hunk 3]
```diff
@@ -325,20 +325,31 @@ def getScores(self, part):
...
+
         if not scores:
             return
 
-        scores = cerealizer.loads(binascii.unhexlify(scores))
         if scores_ext:
-            scores_ext = cerealizer.loads(binascii.unhexlify(scores_ext))
+            try:
+                scores_ext = cerealizer.loads(binascii.unhexlify(scores_ext))
+            except:
```
Comment (Linkid) [inline fofix/game/song/song.py]: You should specify the exception (use `Exception` instead).

**Comment by Linkid** [comment]
I quickly read the code, but didn't test it yet.

File: setup.py [hunk 4]
```diff
@@ -26,18 +26,84 @@
...
+        if not lst:
+            lst = list()
+            outs[subtarg] = lst
+        lst.append(f)
+    return [(k,outs[k]) for k in sorted(outs.keys())]
+
+
+# Hack to fix "No such file or directory: 'numpy-atlas.dll'".
+# https://stackoverflow.com/questions/36191770/py2exe-errno-2-no-such-file-or-directory-numpy-atlas-dll
+if os.name == 'nt':
```
Comment (Linkid) [inline setup.py]: You can put this block within the block [L111](#diff-2eeaed663bd0d25b7e608891384b7298R111).

File: .gitignore (hunk 1 above)
Comment (wrzwicky) [reply .gitignore]: Same change, it just got caught in two different branches. Git is smart enough to merge them.

File: fofix/game/song/song.py (hunk 2 above)
Comment (wrzwicky) [reply fofix/game/song/song.py]: In order to put an exception there, I need to look up the current top of the exception class hiercarchy. But I really don't care what that is, I just want to trap everything. Thus the unqualified "except".

File: fofix/game/song/song.py (hunk 3 above)
Comment (wrzwicky) [reply fofix/game/song/song.py]: As above.

File: setup.py (hunk 4 above)
Comment (wrzwicky) [reply setup.py]: Technically true, but it's not good organization to hide a hack inside the normal program flow. Also, that hack needs to execute before any of the normal setup-building begins.

File: .gitignore (hunk 1 above)
Comment (Linkid) [reply .gitignore]: Nope, git will not know that there is another line later that will do that too.

File: fofix/game/song/song.py (hunk 2 above)
Comment (Linkid) [reply fofix/game/song/song.py]: Please, name your exception. As it is written in the doc, you can either specify the exception or just use `Exception` (this is what you need, if I understand your issue).

File: .gitignore (hunk 1 above)
Comment (wrzwicky) [reply .gitignore]: When you merge, git sees that nothing has changed and quietly merges the files. There's no conflict here.

File: fofix/game/song/song.py (hunk 2 above)
Comment (wrzwicky) [reply fofix/game/song/song.py]: Turns out you are correct, but one should not use Exception. Here's the reason:
https://docs.python.org/2/howto/doanddont.html#except

**Comment by wrzwicky** [comment]
Update complete (setup.py will not be changed), please re-review.

**Comment by Linkid** [comment]
Merged.

**Pull request merged**